core/data/src/main/java/kr/boostcamp_2024/course/data/repository/QuizRepositoryImpl.kt

```diff
@@ -185,43 +174,35 @@ class QuizRepositoryImpl @Inject constructor(
         }
     }
 
-    override fun observeRealTimeQuiz(quizId: String): Flow<Result<RealTimeQuiz>> = callbackFlow {
+    override fun observeRealTimeQuiz(quizId: String): Flow<RealTimeQuiz> = callbackFlow {
         val quizDocument = quizCollectionRef.document(quizId)
         val listener = quizDocument.addSnapshotListener { documentSnapshot, error ->
-            runCatching {
-                if (error != null) {
-                    close(error)
-                }
-
-                if (documentSnapshot?.exists() == true) {
-                    val response = documentSnapshot.toObject(RealTimeQuizDTO::class.java)?.toVO(quizId)
-                    trySend(Result.success(requireNotNull(response)))
-                } else {
-                    throw Exception("문서가 존재하지 않습니다")
-                }
-            }.onFailure { exception ->
-                trySend(Result.failure(exception))
+            if (error != null) {
+                close(error)
+            }
+            if (documentSnapshot?.exists() == true) {
+                val response = documentSnapshot.toObject(RealTimeQuizDTO::class.java)?.toVO(quizId)
+                trySend(requireNotNull(response))
+            } else {
+                throw Exception("문서가 존재하지 않습니다")
             }
```

Comment:
```kotlin
if (documentSnapshot?.exists() == true) {
    val response = documentSnapshot.toObject(RealTimeQuizDTO::class.java)?.toVO(quizId)
    trySend(requireNotNull(response))
} else {
    throw Exception("문서가 존재하지 않습니다")
}
```
해당 코드에서 발생되는 에러가 제대로 처리되지 않을 것 같아요! 위 부분을 추가로 `try-catch`로 감싸는 게 좋을 것 같습니다! 그리고 `throw Exception("..")`만을 하는 것보다 `close(Exception("...")`를 사용해 에러가 발생했을 때 채널이 닫히고, 리스너가 제거되도록 하는 것이 좋을 것 같습니다! 🙂
```suggestion
            if (error != null) {
                close(error)
            }
            try {
                if (documentSnapshot?.exists() == true) {
                    val response = documentSnapshot.toObject(RealTimeQuizDTO::class.java)?.toVO(quizId)
                    trySend(requireNotNull(response))
                } else {
                    throw Exception("문서가 존재하지 않습니다")
                }
            } catch (exception: Exception) {
                close(exception)
            }
```

Reply:
[03b1045](https://github.com/boostcampwm-2024/and02_weQuiz/pull/206/commits/03b10459c146ec997528c14e570f80863cf11ca2) 😁